src/Confluent.RestClient/IConfluentClient.cs

```diff
@@ -89,6 +164,21 @@ public interface IConfluentClient : IDisposable
             int? maxBytes = null);
 
         /// <summary>
+        /// Consume messages from a topic as binary (base 64 encoded string). 
+        /// If the consumer is not yet subscribed to the topic, this adds it as a subscriber, possibly causing a consumer rebalance
+        /// </summary>
+        /// <param name="consumerInstance">Consumer instance</param>
+        /// <param name="topic">The topic to consume messages from</param>
+        /// <param name="cancellationToken">The cancellation token to cancel the operation</param>
+        /// <param name="maxBytes">The maximum number of bytes of un-encoded keys and values that should be included in the response</param>
+        /// <returns>List of Binary messages</returns>
+        Task<ConfluentResponse<List<BinaryMessage>>> ConsumeAsBinaryAsync(
+            ConsumerInstance consumerInstance,
+            string topic,
+            CancellationToken cancellationToken,
+            int? maxBytes = null);
```

Comment:
Let's have CancellationToken cancellationToken as a last parameter.

This implies that we'll have 4 methods to replace this.

```C#
        Task<ConfluentResponse<List<BinaryMessage>>> ConsumeAsBinaryAsync(
            ConsumerInstance consumerInstance,
            string topic);

        Task<ConfluentResponse<List<BinaryMessage>>> ConsumeAsBinaryAsync(
            ConsumerInstance consumerInstance,
            string topic,
            CancellationToken cancellationToken);

        Task<ConfluentResponse<List<BinaryMessage>>> ConsumeAsBinaryAsync(
            ConsumerInstance consumerInstance,
            string topic,
            int maxBytes);

        Task<ConfluentResponse<List<BinaryMessage>>> ConsumeAsBinaryAsync(
            ConsumerInstance consumerInstance,
            string topic,
            int maxBytes,
            CancellationToken cancellationToken);
```

In the actual implementation (i.e. `ConfluentClient`) we can have all the methods calling to one private helper which is already there.

```C#
        public async Task<ConfluentResponse<List<BinaryMessage>>> ConsumeAsBinaryAsync(
            ConsumerInstance consumerInstance,
            string topic,
            CancellationToken cancellationToken,
            int? maxBytes = null)
        {
            string requestUri = BuildConsumeRequestUri(topic, maxBytes);

            // ...
        }
```